pkg/controller/install/deployment.go

```diff
@@ -59,13 +61,25 @@ func (c DeploymentInitializerFuncChain) Apply(deployment *appsv1.Deployment) (er
 // the given context.
 type DeploymentInitializerBuilderFunc func(owner ownerutil.Owner) DeploymentInitializerFunc
 
-func NewStrategyDeploymentInstaller(strategyClient wrappers.InstallStrategyDeploymentInterface, templateAnnotations map[string]string, owner ownerutil.Owner, previousStrategy Strategy, initializers DeploymentInitializerFuncChain) StrategyInstaller {
+func NewStrategyDeploymentInstaller(strategyClient wrappers.InstallStrategyDeploymentInterface, templateAnnotations map[string]string, owner ownerutil.Owner, previousStrategy Strategy, initializers DeploymentInitializerFuncChain, apiServiceDescriptions []v1alpha1.APIServiceDescription, webhookDescriptions []v1alpha1.WebhookDescription) StrategyInstaller {
+	apiDescs := make([]caResource, len(apiServiceDescriptions))
+	for i := range apiServiceDescriptions {
+		apiDescs[i] = &apiServiceDescriptionsWithCAPEM{apiServiceDescriptions[i], []byte{}}
+	}
+
+	webhookDescs := make([]caResource, len(webhookDescriptions))
+	for i := range webhookDescs {
+		webhookDescs[i] = &webhookDescriptionWithCAPEM{webhookDescriptions[i], []byte{}}
+	}
+
 	return &StrategyDeploymentInstaller{
-		strategyClient:      strategyClient,
-		owner:               owner,
-		previousStrategy:    previousStrategy,
-		templateAnnotations: templateAnnotations,
-		initializers:        initializers,
+		strategyClient:         strategyClient,
+		owner:                  owner,
+		previousStrategy:       previousStrategy,
+		templateAnnotations:    templateAnnotations,
+		initializers:           initializers,
+		apiServiceDescriptions: apiDescs,
```

Comment:
refactor suggestion: is there a way we can model the apidescriptions/webhookdescriptions as just more `initializer` functions?